python/ray/serve/_private/replica.py

```diff
@@ -432,25 +432,28 @@ async def invoke_single(self, request_item: Query) -> Tuple[Any, bool]:
         success = True
         try:
             runner_method = self.get_runner_method(request_item)
-            method_to_call = sync_to_async(runner_method)
-            result = None
-            if len(inspect.signature(runner_method).parameters) > 0:
-                result = await method_to_call(*args, **kwargs)
+
+            # Small edge case to allow defining ingress deployments whose
+            # __call__ method takes no arguments.
+            if (
+                len(args) == 1
+                and isinstance(args[0], starlette.requests.Request)
+                and len(inspect.signature(runner_method).parameters) == 0
+            ):
+                args = tuple()
+                kwargs = dict()
+
+            # If the callable is a sync method, run it in the asyncio loop's
+            # default executor.
+            if inspect.iscoroutinefunction(runner_method):
```

Comment:
Why would we want to differentiate b/w sync/async here? 

There could also be poorly implemented async method that might have CPU-bound prologue that would be blocking the event-loop before yielding it.